File: my-folder/0045-jump-game-ii/solution.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dest = len(nums) - 1
        l = 0 
        r = 0
        minsteps = 0
        maxtravel = 0
        while (r < dest):
            for l in range(l, r + 1):
                maxtravel = max(nums[l] + l, maxtravel)

            l = r + 1
            r = maxtravel
            minsteps += 1

        return minsteps
         

            




        

File: my-folder/0045-jump-game-ii/solution.java
class Solution {
    public int jump(int[] nums) {
        int left = 0; 
        int right = 0; 
        int steps = 0;
        while(right < nums.length - 1){
            int maxtravel = 0;
           for(int i = left; i < right + 1; i++){
                maxtravel = Math.max(maxtravel, i + nums[i]);
           }

           left = right + 1;
           right = maxtravel;
           steps += 1;
        }

        return steps;
    }
}
